File: iteration2/CSE3063F21P1_GRP16/src/main/java/Models/E1ReTakeCourse.java
package Models;


import java.util.ArrayList;

public class E1ReTakeCourse  implements CourseEvents {
	
	
private ArrayList<Student> students;


public ArrayList<Student> getStudents() {
	return students;
}

public  E1ReTakeCourse() {
	students=new ArrayList<>();
	
}

@Override
public String toString() {

	String s="    Total Number of Student who taking the course again: "+students.size();
	if(students.size()>0)
		s+="\n     -List Of Students:\n";
	else
		s+="\n";

	for(int i=0;i<students.size();i++) {
		s+="      *"+students.get(i).getFirstName()+" "+students.get(i).getLastName()+"\n";



	}
	return s;
}

}

File: iteration3/CSE3063F21P1_GRP16/models/E1ReTakeCourse.py
from typing import List

from CourseEvents import CourseEvents

from Student import Student


class E1ReTakeCourse(CourseEvents):
    def __init__(self):
        self._students = []

    def getStudents(self):
        return self._students

    def toString(self):
        s = "    Total Number of Student who taking the course again: {}\n".format(len(self._students))
        if len(self._students) > 0:
            s += "     -List Of students:\n"
        else:
            s += "\n"

        for i in range(len(self._students)):
            s += "      *{} {}\n".format(self._students[i].getFirstName(), self._students[i].getLastName())

        return s
